pkg/common/configs/configvalidator.go

```diff
@@ -595,6 +595,28 @@ func checkLimits(limits []Limit, obj string, queue *QueueConfig) error {
 	return nil
 }
 
+func checkLimitsStructure(partitionConfig *PartitionConfig) error {
+	partitionLimits := partitionConfig.Limits
+	rootQueue := partitionConfig.Queues[0]
+
+	if len(partitionConfig.Queues) < 1 || strings.ToLower(partitionConfig.Queues[0].Name) != RootQueue {
+		return fmt.Errorf("top queue is not root")
+	}
+
+	if len(partitionLimits) > 0 && len(rootQueue.Limits) > 0 {
+		return fmt.Errorf("partition limits and root queue limits both exist")
+	}
+
+	// if root queue limits not defined, apply partition limits
+	if len(partitionLimits) > 0 && len(rootQueue.Limits) == 0 {
+		partitionConfig.Queues[0].Limits = partitionLimits
+		// clear partition limits
+		partitionConfig.Limits = nil
```

Comment:
Could you add comment to https://github.com/apache/yunikorn-core/blob/master/pkg/common/configs/config.go#L52 to reminder readers that the field could be empty. Also, we should honor the root-level limits.

Reply:
Or we should make sure that partition limits are always equal to root limits? We do NOT undefine the partition limits. Instead, we allow both limits if they are identical.

Reply:
> Instead, we allow both limits if they are identical.

Sounds good, I'll adopt your approach. Many thanks !